### ISSUE TITLE
[Bug] status update stops working 

### ISSUE THREAD

**Issue by jappish84** [JoshuaMulliken/ha-wyzeapi#297]
**Describe the bug**
I have 5 wyze v2 and 1 wyze v3 cams connected, and while controlling the cameras on and off through service calls works reliably, the status update does not. 

I have automations that trigger cameras on/off when certain conditions are met, and the cameras respond every time it seems, but the status gets stuck in "off" state after a few days (seems to only get stuck in "off" state) 

I have followed the steps to enable logging for the integration, and manually toggling the switches does not generate any log entry. Trying to toggle a cam switch entity that's 'on' but shows 'off' simply just falls back to 'off'  state, even though the cam is on. 

Reloading the integrations solves the issue temporarily. 

**To Reproduce**

1. Wait
2. See above

**Expected behavior**
Working state update 

**System configuration**
System: hassio, VM
HA Version: 2021.11.04
WyzeApi Version: 2021.10.2

**home-assistant.log**

Can't get any relevant logs

**Comment by JoshuaMulliken** [comment]
@yoinx this seems to be a pattern with the new method. Could requests be failing and then locking up a slot or something?

**Comment by yoinx** [comment]
@JoshuaMulliken it could be. I meant to increase the logging output in the update manager to be able to diagnose issues like this, but I haven't had a chance yet.

I will say that toggling the entry though bypasses the update manager, so it's a little odd that there's no output from that.

**Comment by yoinx** [comment]
@jappish84  I'm looking into this still. Found something interesting... Just out of curiosity is one of your devices a lock? If so, does it also have a keypad?

**Comment by jappish84** [comment]
Thanks @yoinx really appreciate the work you all are putting into it.

No locks unfortunately, just cams, five v2 cams, and one v3 cam, that's all that's connected to the account.

**Comment by yoinx** [comment]
Ah, thanks. Just just be a coincidence then that mines failing on a call for the lock.

Just because I'm all about full transparency... What I'm seeing is that after awhile (no real specific time) I get a 502/Bad Gateway, which then cascades up and kills the update manager. We're working on adding in some error handling for that which should hopefully prevent this from happening.

**Comment by jappish84** [comment]
Sounds great, just let me know if there if you need any further info or additional testing 👍

**Comment by yoinx** [comment]
This should be resolved in the new beta: 2021.12.1b1

**Comment by xdreamwalker** [comment]
I'm new to HA and came across your Wyze integration. I've had this problem, where status is not updated from Wyze. I have plugs and locks (with keypads). After an undetermined amount of time if a lock is manually changed the status in HA is not updated.
Being new and not knowing much at all, can someone help me figure out out to get the beta channel in HACS?
![image](https://user-images.githubusercontent.com/24192512/145031684-a89ea591-3256-4f2d-805f-38ffa7d00d38.png)
I've selected the beta checkbox, but am still only presented with 2021.11.1

**Comment by yoinx** [comment]
We had to remove the beta I referenced above due to login issues introduced with it (presumably from what I did to fix this issue). So I have to dig into that before we put out a new beta.

**Comment by JoshuaMulliken** [comment]
Btw local control for bulbs coming soon. I have not investigated if that is implemented for the switches but you can see my progress on it here: https://mulliken.net/blog/reverse-engineering-iot-communication

**Comment by JoshuaMulliken** [comment]
Seems similar to the issue in #322 and #321. Might also be the reason why the (most recent and accurate) IP is not loaded in #315

**Comment by tggman** [comment]
> Seems similar to the issue in #322 and #321. Might also be the reason why the (most recent and accurate) IP is not loaded in #315

A few of my color bulbs still fail to respond to HA commands for days at a time while still functioning as expected in the Wyze app. During that time, their state typically reverts back to their actual state a few seconds after an attempt is made to change their state in HA. 

Because the bulbs were still available in HA, I was able to see and confirm that the IP address listed in the bulb's HA attributes matched their actual value (as per my router and the Wyze App).

I thought about re-opening #315 but in the end figured it would be redundant relative to  subsequent open issues.

**Comment by JoshuaMulliken** [comment]
@tggman it may also be because the encryption value is not correct. I am not sure if they rotate that value.

**Comment by JoshuaMulliken** [comment]
It seems to be stable for me. Let me know how it is for all of you

**Comment by jappish84** [comment]
I find it beeing stable now! Great work! Thanks!

**Comment by bobloadmire** [comment]
I'm having this issue now with my wyze light strips. I can turn them on and off in the app and everything is up to date, but if I turn on or off the lights with the physical button, HA doesn't update to match the status.

**Comment by brg468** [comment]
Does the app reflect the correct state when you use the physical button? I don’t know that I ever tried it..

**Comment by bobloadmire** [comment]
> Does the app reflect the correct state when you use the physical button? I don’t know that I ever tried it..

yes the app does update, although it's a few seconds delayed. I was under the impression since the strips are under local control it would pretty much be instant?

**Comment by brg468** [comment]
I’m pretty sure the Wyze app doesn’t use local control at all, and this integration only uses local control for commands, the state is still polled on an interval.

**Comment by bobloadmire** [comment]
well crap

**Comment by jk111** [comment]
My color bulbs status in HA only works for an hour or so after reloading the wyze service. Been like this for a few months now. I'll try to look at the logs to see if I can diagnose.

**Comment by JoshuaMulliken** [comment]
Shit looks like they are tightening the screws...

**Comment by prnewman** [comment]
@chrisvball Here's the [RTSP firmware](https://support.wyze.com/hc/en-us/articles/360026245231-Wyze-Cam-RTSP). According to doc: "Please note that the Wyze Cam Pan firmware file is not compatible with Wyze Cam Pan v2."

**Comment by manofpants** [comment]
Hi all. I'm seeing the same thing on my Wyze thermostats and bulbs. I have schedules set in the Wyze app to change the thermos a few times a day and they will get stuck on on and not update for a day or two. I tried setting up an automation, but that actually seemed to make it worse because of too many polls probably.
![image](https://user-images.githubusercontent.com/8443799/180815617-d8253843-2d87-4c8c-bf57-dab54e31a13d.png)

**Comment by widewing** [comment]
Hi I have implement a fix to this bug. I tested several days and it seems working correctly now. @JoshuaMulliken can you take a look? https://github.com/JoshuaMulliken/wyzeapy/pull/54

**Comment by JoshuaMulliken** [comment]
I have published the fix from @widewing !!

Thank you for your contribution!
